src/main/kotlin/com/fasterxml/jackson/module/kotlin/support/ValueClassSupport.kt

```diff
@@ -0,0 +1,42 @@
+package com.fasterxml.jackson.module.kotlin.support
+
+import kotlin.reflect.KClass
+import kotlin.reflect.KProperty1
+
+/**
+ * This class only support kotlin value class
+ */
+public object ValueClassSupport {
+
+    /**
+     * get boxed value of a **`value class`**.
+     * if the receiver is not a `value class`, throw [UnsupportedOperationException]
+     */
+    public val <T : Any> T.boxedValue: Any?
+        @Suppress("UNCHECKED_CAST")
+        get() = if (this::class.isValue) {
+            (this::class as KClass<T>).boxedProperty.get(this)
+        } else {
+            throw UnsupportedOperationException("$this is not a value class")
```

Comment:
I think Returning `this` is may be better than throw UnsupportedOperationException.
What do you think about this? @k163377